### PR TITLE
Initial basic C API for EAR manipulation

### PR DESCRIPTION
**Pull request by thomas-fossati**
This PR partially addresses [1]

[1] https://github.com/CCC-Attestation/attested-tls-poc/issues/17

Signed-off-by: Thomas Fossati <thomas.fossati@arm.com>

### CODE REVIEW

**Comment by thomas-fossati** [comment]
cc @paulhowardarm

File: src/ear.c [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    (void)snprintf(e, sizeof e, "unknown eat_profile \"%s\"", eat_profile);
+    goto err;
+  }
+
+  const char *status = jwt_get_grant(ear->jwt, "ear.status");
+
+  if (status == NULL) {
+    (void)snprintf(e, sizeof e, "missing ear.status claim");
+    goto err;
+  }
```
Comment (thomas-fossati) [inline src/ear.c]: leftover, this is now taken care of by the `ear_get_status` API

```suggestion
```

**Review by setrofim — APPROVED** [review]
The fact that this expects an `ear.status` at the top level suggests that hish implmented against a now-outdated version of EAR draft? Other than that, looks good!

**Comment by thomas-fossati** [comment]
> The fact that this expects an `ear.status` at the top level suggests that hish implmented against a now-outdated version of EAR draft? Other than that, looks good!

Correct, I've coded against the old spec:
```c
#define EAR_PROFILE_2022 "tag:github.com,2022:veraison/ear"
```

**Pull request merged**